src/Client/Core/PurgeResult.cs

```diff
@@ -18,9 +18,29 @@ public PurgeResult(int count)
         this.PurgedInstanceCount = count;
     }
 
+    /// <summary>
+    /// Initializes a new instance of the <see cref="PurgeResult" /> class.
+    /// </summary>
+    /// <param name="count">The number of instances deleted.</param>
+    /// <param name="isComplete">A value indicating whether the purge operation is complete. 
+    /// If true, the purge operation is complete. All instances were purged.
+    /// If false, not all instances were purged. Please purge again.
+    /// If null, default to legacy purge behavior. All instances were purged.</param>
+    public PurgeResult(int count, bool? isComplete)
+        : this(count)
+    {
+        this.IsComplete = isComplete;
+    }
+
     /// <summary>
     /// Gets the number of purged instances.
     /// </summary>
     /// <value>The number of purged instances.</value>
     public int PurgedInstanceCount { get; }
+
+    /// <summary>
+    /// Gets a value indicating whether the purge operation is complete.
+    /// </summary>
+    /// <value>A value indicating whether the purge operation is complete.</value>
+    public bool? IsComplete { get; }
```

Comment:
Do we actually need to expose this to the caller? We could leave it part of the gRPC contract only, and the implementation of purge will make repeated calls until purge is complete.

Reply:
i think it is needed for customers who want to do partial purges themselves (ex. do partial purges following their own mechanism, adding custom loggings, quit early, etc)

Reply:
i can update to add purgeallinstances to do loop internally as alternative approach

Reply:
Depending on the backend, purge can be prohibitively expensive, so I don't think we should automate this loop.